certbot/main.py

```diff
@@ -653,34 +673,26 @@ def obtain_cert(config, plugins, lineage=None):
         raise
     le_client = _init_le_client(config, auth, installer)
 
-    # SHOWTIME: Possibly obtain/renew a cert, and set action to renew | newcert | reinstall
-    if config.csr is None: # the common case
-        domains, certname = _find_domains_or_certname(config, installer)
-        action, _ = _auth_from_available(le_client, config, domains, certname, lineage)
-    else:
-        assert lineage is None, "Did not expect a CSR with a RenewableCert"
-        _csr_obtain_cert(config, le_client)
-        action = "newcert"
+    if config.csr:
+        cert_path, fullchain_path = _csr_obtain_cert(config, le_client)
+        _report_new_cert(config, cert_path, fullchain_path)
```

Comment:
For both this and renaming `_csr_obtain_cert` to `_csr_certonly`, that depends on how we're conceptualizing `_csr_obtain_cert`. I was thinking of it as analogous to `_get_and_save_cert`, whereas it seems like you're thinking of it as more analogous to `certonly`. If we were going to move these lines into `_csr_obtain_cert`, I'd actually also move it above initializing `le_client` and just fully split the paths immediately. But instead of doing that, maybe we could rename to `_csr_get_and_save_cert`. Would that make it more clear? Or are you still worried about the `None` thing? I can also put a comment in `_report_new_cert` that `cert_path` and `fullchain_path` will be `None` if we're calling from a dry run.